### DISCUSSION TITLE
5-axis gantry controller

### DISCUSSION THREAD

**Discussion by tobjy** [grblHAL/core#414] [Q&A] [answered]
I am building a 5-axis gantry for laser cutting. The mechanics is going good so far, but I am struggling selecting a controller board and software. I want to get the electronics running before I build everything together. 

Best choice so far seems to be grblHAL for the FYSETC S6 V2 board. The web-builder does however not allow me to select 5 axis, it is fixed to 3. Does anyone know why? Does the board not support true 5-axis synchronized movements? Or is it because none of the out-of-the-box kinematic models supports more than 3 axis?

At a first glance of the source code, it seems that only a few kinematic models are available "out-of-the-box". I suppose I will need to code my own kinematics, right?

**Comment by terjeio** [comment, marked as answer]
> The web-builder does however not allow me to select 5 axis, it is fixed to 3. Does anyone know why?

The board definition file (not the board map) is incorrect, max. number of axes/motors is 6, FYI auto squaring is not supported (due to limit switch inputs not beeing assigned).

> Does the board not support true 5-axis synchronized movements? 

It supports up to 6 axes if configured for one motor per axis.

>  I suppose I will need to code my own kinematics, right?

If you need a different one than those already supported then yes.

**Reply by tobjy** [reply]
Thank you very much for confirming that. If I ever complete this project, I will share the code, who knows, might help someone else. 
God Jul och gott nytt år!

**Reply by terjeio** [reply]
> God Jul och gott nytt år!

Takk det samme!

I'll update the board definition file in the next commit.
